test/verify/check-machines

```diff
@@ -3435,6 +3436,74 @@ class TestMachines(MachineCase, StorageHelpers, NetworkHelpers):
         b.click('#deactivate-network-test_network4-{0}'.format(connectionName)) # Deactivate transient network
         self.waitNetworkRow("test_network4", connectionName, False) # Check it's not present after deactivation
 
+    def testSnapshots(self):
+        # Checks if difference of @time1 and @time2 is not greater than @difference (in seconds)
+        def checkTimeDiff(time1, time2, difference):
+            tmp = time2.split(' ') # split "Today 13:13:13" into day and time
+            if not tmp[1].startswith('00:') and not tmp[0] == "Today":
+                return False
+
+            diff = datetime.strptime(time1, '%H:%M:%S') - datetime.strptime(tmp[1] + tmp[2], '%I:%M%p')
+            return -difference < diff.total_seconds() < difference
+
+        b = self.browser
+        m = self.machine
+
+        self.startVm("subVmTest1")
+
+        snapshots_present = m.execute("gdbus introspect --xml --system --dest org.libvirt --object-path /org/libvirt/QEMU | grep snapshot || true")
+        if not snapshots_present.strip(): # libvirt-dbus doesn't have snapshot APIs
+            return
```

Comment:
This is brittle, as it will silently skip the test if the snapshot API breaks in the future. We want to control this more tightly. I. e. please enumerate the OSes where snapshots *don't* work. This will be a long list at first of course, but over time we can just remove OSes where it starts to work, and when adding new OSes we won't forget to update the list.